workers/unity/Packages/com.improbable.gdk.deploymentlauncher/.DeploymentLauncher/Commands/Create.cs

```diff
@@ -0,0 +1,191 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Security.Cryptography;
+using Google.Protobuf.WellKnownTypes;
+using Improbable.SpatialOS.Deployment.V1Alpha1;
+using Improbable.SpatialOS.PlayerAuth.V2Alpha1;
+using Improbable.SpatialOS.Snapshot.V1Alpha1;
+using Newtonsoft.Json.Linq;
+
+namespace Improbable.Gdk.DeploymentLauncher.Commands
+{
+    public static class Create
+    {
+        public static int CreateDeployment(Options.Create options)
+        {
+            return CreateDeploymentInternal(options, opts => File.ReadAllText(opts.LaunchJsonPath));
+        }
+
+        public static int CreateSimulatedPlayerDeployment(Options.CreateSimulated options)
+        {
+            return CreateDeploymentInternal(options, opts => ModifySimulatedPlayerLaunchJson(options));
+        }
+
+        private static int CreateDeploymentInternal<TOptions>(TOptions options, Func<TOptions, string> getLaunchConfigJson)
+            where TOptions : Options.Create
+        {
+            var snapshotServiceClient = SnapshotServiceClient.Create();
+            var deploymentServiceClient = DeploymentServiceClient.Create();
+
+            try
+            {
+                var deployment = new Deployment
+                {
+                    AssemblyId = options.AssemblyName,
+                    LaunchConfig = new LaunchConfig
+                    {
+                        ConfigJson = getLaunchConfigJson(options)
+                    },
+                    Name = options.DeploymentName,
+                    ProjectName = options.ProjectName,
+                    RegionCode = options.Region
+                };
+
+                if (options.SnapshotPath != null)
+                {
+                    var snapshotId = UploadSnapshot(snapshotServiceClient, options.SnapshotPath, options.ProjectName,
+                        options.DeploymentName);
+
+                    if (string.IsNullOrEmpty(snapshotId))
```

Comment:
Anything useful that could be printed here?

Reply:
The two cases where we return `null` from `UploadSnapshot()` we've already written out more specific errors. We could return `Result<string,string>` and unwrap accordingly so we handle the error here.